pandas/plotting/_matplotlib/core.py

```diff
@@ -124,13 +126,34 @@ def __init__(
         table=False,
         layout=None,
         include_bool=False,
+        column: IndexLabel | None = None,
         **kwds,
     ):
 
         import matplotlib.pyplot as plt
 
         self.data = data
-        self.by = by
+        self.by = com.maybe_make_list(by)
+
+        # Assign the rest of columns into self.columns if by is explicitly defined
+        # while column is not, so as to keep the same behaviour with current df.hist
+        # or df.boxplot.
+        if self.by and column is None:
+            self.columns = [
+                col
+                for col in data.columns
+                if col not in self.by and is_numeric_dtype(data[col])
+            ]
+        else:
+            self.columns = com.maybe_make_list(column)
```

Comment:
yeah, this is to be aligned with `DataFrame.boxplot` and `DataFrame.hist` plot (e.g. https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.boxplot.html) , in current APIs, both provide the argument `column` to sub-select for visualization. 

You are very right that we could also first sub-select columns and apply with plotting function. But since one of the purposes of having this PR is to pave the way for deprecating those `DataFrame.boxplot` and `DataFrame.hist`, so I added here. What's your thoughts?

Reply:
This makes sense. Let's keep it for now, so we can remove the duplicate code without breaking users code. But if you want to create an issue to deprecate the `column` parameter in the future, that would be great.